src/extractor/worker.ts

```diff
@@ -2,26 +2,26 @@ import { resolve, extname } from 'path';
 import { Worker, isMainThread, parentPort } from 'worker_threads';
 import { readFile } from 'fs/promises';
 
+// TODO: this solution won't handle new integrations and it will need a slight tweaking before adding new ones
+import internalExtractor from './presets/react';
 import { loadModule } from '../utils/moduleLoader';
 import { type Deferred, createDeferred } from '../utils/deferred';
 
-export type WorkerParams = { extractor: string; file: string };
+export type WorkerParams = { extractor?: string; file: string };
 
 const IS_TS_NODE = extname(__filename) === '.ts';
 
 // --- Worker functions
 
-let loadedExtractor: string;
+let loadedExtractor: string | undefined | symbol = Symbol('unloaded');
 let extractor: Function;
 
-async function loadExtractor(extractorScript: string) {
-  const mdl = await loadModule(extractorScript);
-  extractor = mdl.default;
-}
-
 async function handleJob(args: WorkerParams) {
   if (loadedExtractor !== args.extractor) {
-    await loadExtractor(args.extractor);
+    loadedExtractor = args.extractor;
+    extractor = args.extractor
+      ? await loadModule(args.extractor).then((mdl) => mdl.default)
+      : internalExtractor;
```

Comment:
Do I understand it correctly that now if you don't specify the extractor it will automatically use "react", since thats the only one currently?

Reply:
Yes, that is correct. As I said, it's wired this way because of some tweaks I need to do to ensure files with comments only are processed.

Reply:
👍